app/assets/javascripts/discourse/app/components/site-header.js

```diff
@@ -40,6 +40,20 @@ const SiteHeaderComponent = MountWidget.extend(
       this.queueRerender();
     },
 
+    @observes("site.narrowDesktopView")
+    narrowDesktopViewChanged() {
+      if (
+        this.siteSettings.enable_experimental_sidebar_hamburger &&
+        (!this.sidebarEnabled || this.site.narrowDesktopView)
+      ) {
+        this.appEvents.on(
```

Comment:
I think my comment got lost but do we need to deregister this?

Reply:
You had comment about another one which I removed completely :) 

That one is deregistered in willDestroy block
https://github.com/discourse/discourse/blob/e473302a0fa4db853b7e7c5d424ab7cd5ad72f2f/app/assets/javascripts/discourse/app/components/site-header.js#L330